### ISSUE TITLE
[Pull Request] A feature flag to disable date tab.

### ISSUE THREAD

**Issue by farhaanbukhsh** [openedx/platform-roadmap#392]
### 1. Is there an existing issue for this?

- [X] I have searched the existing issues

### 2. What new feature or functionality would you like to request?

There is no way to disable tabs on the course page 

![image](https://github.com/user-attachments/assets/bc56bd34-9874-4be9-b5dc-9adf5ee37052)

We need a way to disable individual Tabs. We started this with 'Dates Tab', this feature flag helps to hide 'Dates Tab' throughout the instance. For this, we have worked on a PR:

https://github.com/openedx/edx-platform/pull/34511

### 3. What product area does this feature affect?

Courseware

### 4. Please describe the scope of the feature

This affects the tabs that appear on the courseware LMS page.

### 5. Please describe why you see a need for this feature

This feature flag helps to customize the instance and helps to convey the information more realistically.

### 6. Please describe the potential impact and/or value of this feature

As mentioned above it helps in customizing the instance.

### 7. Please provide 2-3 use cases and/or user stories in support of this feature

Helps to hide the dates since there are times when it is not relevant to the organization using it.

**Comment by marcotuts** [comment]
**Site Level Feature Flag: Not Recommended for Merge** 
I don't know how commonly an instance-wide Dates tab disabling would be, so I don't know (as with the other Progress Tab customization PR) if this should be merged upstream as a feature flag versus just a site-level front-end slot customization done for a specific client.  

This is not a strong stance to be clear, but I don't know if we expect this will be commonly used at the site level. 

____
**Course Level Pages & Resources Card + Disable Option: Recommended for Merge (?!?)** 
Separately, I did wonder if Dates should be added to the Pages & Resources view of Studio, since both Discussions and Progress are visible here and can be enabled + disabled from this view.  (Shown below in the attachment). I don't know what kinds of settings we would want to expose in the Dates area if any, but perhaps like the progress page it is rather empty other than being able to be enabled / disabled. 

<img width="1038" alt="image" src="https://github.com/user-attachments/assets/d12ac79a-dbff-4d3c-8f27-44a54bc7fbd6">


Is it strange that the case for making this a site-level feature flag seems weak, but that the experience completeness + customization balance of having Dates be added for each course to decide whether they want to turn it off  (which would be VERY infrequently used, as with turning off the progress page) is somehow a more compelling case in my mind? It feels a bit strange to me but somehow I think if I'd seen this as a proposal to "Match Dates page customization for Courses" and enable similar functionality as Progress / Discussion / Wiki / Textbooks / Static Pages, it would have been a thumbs up from me.

**Comment by farhaanbukhsh** [comment]
> Is it strange that the case for making this a site-level feature flag seems weak, but that the experience completeness + customization balance of having Dates be added for each course to decide whether they want to turn it off (which would be VERY infrequently used, as with turning off the progress page) is somehow a more compelling case in my mind?

Actually @marcotuts  you are right on this part because intuitively even I thought that it makes more sense for the course to have a setting where we can turn the date tab on and off and it was very human to think that. But hear me out on this the argument is if I want to disable a date tab on an instance I would want that behavior for all the courses in a uniform fashion whereas having a date tab for some courses while not having one on another will create a dissonance in the learners.

Now, this might not be true for wiki, discussion or pages since they have functionality and actions attached to them while date is just a way of conveying information.

I am open to suggestions and better way of taking this approach here.

Thanks a lot for the quick revert @marcotuts

**Comment by marcotuts** [comment]
Hi @farhaanbukhsh, the inconsistency you highlight when you enable/disable course tabs within an instance is absolutely an issue, but at edX we found organizations that needed / (or felt they needed) this level of control even if it left learners with not a consistent cross-course experience. 

This is somewhat related to the example I mentioned in the Progress controls PR (quote excerpts is below)- https://github.com/openedx/platform-roadmap/issues/389

> I could imagine an instance having 2-3 types of courses: standard, certification, and capstone. This example course 'type' isn't reflected by an existing setting but in this hypothetical you might have a single project submission in the capstone, and a practice test or two plus a final exam in the certification course type. Across these course 'types' you could imagine wanting to hide different areas of the progress page that feel less relevant, or (as I saw in another pull request - hide the dates tab?).
> 
> In other words perhaps there are mixes of courses where you'd like to hide larger areas of the course but keep them enabled elsewhere, but it still doesn't strike me as common a need to justify upstreaming the more I consider this and how this pattern would be echoed into other parts of the course tools / mobile apps / course-level apps.


**Instance Level**  - It should be possible to more easily control which tabs are shown / hidden by default in my course, and which areas of the Pages & Resources view should be hidden. This won't be common but it allows for the control of hiding all Dates / Progress / Discussion tabs or any combination of these at the instance level. This is most like the PR proposal currently, though I wonder if we should consider what it would look like to configure any of the course tabs / pages & resources areas consistently instead of just 1 feature flag for dates. **Do any of the other tabs have feature flags or configuration options today?** 

**Course Level** - How common should the need be to be able to disable a page at the course level? The Pages & Resources view / UX experience doesn't place a value judgment on whether you should do this but it does show the controls for Discussion, Wiki, Progress, but it is missing Dates currently.  FWIW I'm not sure that I agree that the current pages have functionality/actions attached to them and Dates does not. Dates let you see / action on verification/upgrade / assignment deadlines the same way that the progress page MOSTLY conveys information and also has actions like clicking through to assignments, upgrading, etc. (Discussions + Wiki are for sure very different and have lots of interactivity) This isn't necessarily in scope for this PR but still worth considering at some future point in time.

**Comment by farhaanbukhsh** [comment]
Hello folks :)  Are there any updates on this?

**Comment by farhaanbukhsh** [comment]
Hey @marcotuts @jmakowski1123 can we finalize this? Please let us know if you have any feedback.

**Comment by pdpinch** [comment]
What's your user story for hiding the dates tab @farhaanbukhsh?

We have one, but it might lead to a different approach, more like @marcotuts proposal. 

We have courses that are "archived" but perpetually open. We want to encourage users to drop in at any time to access the materials, but discussions are closed and there's no certificates. For these courses, _all_ the UI elements about dates -- not just the dates tab -- are distracting and confusing and lead to helpdesk tickets. 

Disabling the dates tab would be helpful, but we'd actually like to turn off more dates UI. 

There are, of course, other ways to do what I'm thinking of. I don't bring this up to hijack your proposal, but just to move towards a better articulation of the product goal.

**Comment by Cup0fCoffee** [comment]
@pdpinch Chiming in on behalf of @farhaanbukhsh in regards to the user story.

One of our clients reached out to us with the following request:
> We would like to hide the course date tab as all our courses are currently asynchronous with rolling enrollment and no official start date

In other words, they would like to like a way to "hide date tab for all courses". That is why in the upstream PR -https://github.com/openedx/edx-platform/pull/34511 - we went with the option of introducing a feature flag. All other types of flags would only allow hiding by course or org (afaik). But we are open to a compromise.

We dismissed the following options:
* Making dates tab togglable, like other tabs. - Didn't fit our user story, because that would make course authors remember to do it for all courses, i.e. inconvenient.
* Adding a `CourseWaffleFlag` for toggling the visibility of dates tab. - Isn't global enough. Afaik, the most global you can do with that is per organization.

I guess it could also be a `SiteConfig` option, but afaik, aside from `MFE_CONFIG`, it's being deprecated for all other stuff.

I wonder if we've overlooked anything else.

I understand that downsides of our proposed option, i.e. not flexible (global, can't be overridden), is not aligned with how other tabs work.

> Disabling the dates tab would be helpful, but we'd actually like to turn off more dates UI.

This is also aligns with the requests that we've been receiving, e.g. hiding things like this:

![Image](https://github.com/user-attachments/assets/48b2e64c-1bfa-444f-b5ea-8e6e738541f0)

But it's beyond the scope of this proposal.

**Comment by farhaanbukhsh** [comment]
Hi @pdpinch 

>There are, of course, other ways to do what I'm thinking of. I don't bring this up to hijack your proposal, but just to move towards a better articulation of the product goal.

I am happy that you pitched in, thank you 😄 

I hope @Cup0fCoffee has given you the user story. I am hoping we could come with a way maybe that can benefit everyone.

**Comment by jmakowski1123** [comment]
With this approach:

Making dates tab togglable, like other tabs. - Didn't fit our user story, because that would make course authors remember to do it for all courses, i.e. inconvenient.

If the dates tab is togglable, is it possible to have a bulk "select all courses" option, to get around the inconvenience of having to choose course by course?

**Comment by Cup0fCoffee** [comment]
@jmakowski1123 

> If the dates tab is togglable, is it possible to have a bulk "select all courses" option, to get around the inconvenience of having to choose course by course?

I think that it would be great in general to have an ability (i.e. implemented as a feature in Open edX with UI and stuff) to apply some configuration change to multiple courses at once. If that was available in the platform, I think making dates tab togglable would be an even better solution than we are proposing, because unlike the feature flag, the toggle can be managed by course staff.

**Comment by jmakowski1123** [comment]
> [@jmakowski1123](https://github.com/jmakowski1123)
> 
> > If the dates tab is togglable, is it possible to have a bulk "select all courses" option, to get around the inconvenience of having to choose course by course?
> 
> I think that it would be great in general to have an ability (i.e. implemented as a feature in Open edX with UI and stuff) to apply some configuration change to multiple courses at once. If that was available in the platform, I think making dates tab togglable would be an even better solution than we are proposing, because unlike the feature flag, the toggle can be managed by course staff.

How big of a lift is it to do some discovery/sizing for that?

**Comment by FlorianO** [comment]
I have the same issue as we use the platform for a company internal training so there is no sense in having dates at all.

**Comment by farhaanbukhsh** [comment]
@ali-hugo @cassiezamparini  This is the proposal I was talking about 😄

**Comment by ali-hugo** [comment]
@Cup0fCoffee @farhaanbukhsh 

> How big of a lift is it to do some discovery/sizing for that?

Could one of you answer @jmakowski1123's question above? It looks like some more work would be required here before coming up with a solution that meets all the use cases.

**Comment by Cup0fCoffee** [comment]
@ali-hugo @jmakowski1123 I'll prepare a list of potential solutions and will post it here.

cc. @farhaanbukhsh

**Comment by Cup0fCoffee** [comment]
@jmakowski1123 

> How big of a lift is it to do some discovery/sizing for that?

I think the cornerstone of implementing a feature that would allow applying a change to two or more courses would be designing good UI/UX. This would involve coming to an agreement on the best UX for the users and creating mockups (probably from scratch) for the UI (which would probably have to be a completely new screen or a modal, because I don't think we can reuse any of the existing components). Another question that we would need to find an answer to, is which settings would we allow to apply in this manner? Course settings exist in many places - advanced settings, pages&resources, feature flags, etc. Would we allow all of them to be applied via this UI and if yes, how can we make a homogeneous UI for all these settings? In general, there is not many unknowns, so I would say that it would be a "medium" size discovery.

We would've loved to contribute something like this to the platform. However, we are constrained by the client's resources.

I am keen on finding a compromise. I've reread the discussion, and I think a good compromise would be changing this product proposal and the PR implementation to make the dates tab togglable the same was as the progress tab is right now. The reasons I say that are:
* Marco made a compelling argument for such implementation above.
* This is something that we can propose to our client and it would be in the range of the budget they would be willing to contribute.
* For the use case of our client, they don't need to toggle the dates tab often, so it doesn't sound like a big compromise.
* It aligns with the potential future plans to create UI/UX for applying a change to multiple courses in bulk.

To summarize, do you think if we were to change the product proposal and the PR to just make the dates tab togglable the same way as progress tab is, instead of creating a global feature flag, then we would be able to get this accepted and move forward?

cc. @farhaanbukhsh @ali-hugo

**Comment by ali-hugo** [comment]
@Cup0fCoffee Thanks for the summary. 

> ...do you think if we were to change the product proposal and the PR to just make the dates tab togglable the same way as progress tab is, instead of creating a global feature flag, then we would be able to get this accepted and move forward?

Just want to make sure I understand what you have in mind. Would course staff enable/disable the dates tab from the _Pages and Resources_ page as described by @marcotuts in [this comment](https://github.com/openedx/platform-roadmap/issues/392#issuecomment-2472242465)?

**Comment by Cup0fCoffee** [comment]
@ali-hugo Yes, that is right. Specifically the section of the comment that is titled "**Course Level Pages & Resources Card + Disable Option: Recommended for Merge (?!?)**".

**Comment by ali-hugo** [comment]
@Cup0fCoffee Thank you. This approach gets a +1 from me.
I'll leave it to @jmakowski1123 as the assignee to give the final go-ahead.

**Comment by sarina** [comment]
@jmakowski1123 to take a look per PWG meeting

**Comment by ali-hugo** [comment]
@jmakowski1123 We’re looking for one more 👍 on the change described below:

A **Dates** card should be added to the _Pages and Resources_ page, giving course teams an easy way to enable and disable the **Dates** tab (just like they can already do with **Discussions** and **Progress**):

<img width="1038" height="690" alt="Image" src="https://github.com/user-attachments/assets/3d7d4f17-ecec-4551-925c-982c840aecef" />


@jmakowski1123 Does this get your go-ahead?

**Comment by ali-hugo** [comment]
@farhaanbukhsh @Cup0fCoffee I've given the Product Working Group 1 week to raise any objections to the approach outlined above ([Slack message](https://openedx.slack.com/archives/C057J2D1WU9/p1760006796835729)). If no one objects, we can consider this approved. 

I'll be in touch in a week to let you know the outcome.

**Comment by sarina** [comment]
There's a lot of comment history here I didn't revisit, but the screenshot two comments above looks good to me.

**Comment by jristau1984** [comment]
I am not sure i understand the widespread use case of this,  and didnt see any real reference here.  I know that it would not be leveraged on edx.org, but as a platform feature I have no objection.

I am not a fan of continuing to pile on to the pages and resources page... it will soon become another advanced settings page.  Please ensure this has some ability to disable at the instance level, so it does not have to show.  Thanks!

**Comment by ali-hugo** [comment]
Thanks @sarina and @jristau1984!

> I am not a fan of continuing to pile on to the pages and resources page... it will soon become another advanced settings page.

While I agree with this, I do think it makes sense to have the _dates_ configurations in the same place as the configurations for _progress, discussion, wiki_ etc; since all of these appear as tabs on the same page in Studio, I'd expect their configurations to happen in the same place. I think _Pages and Resources_ makes sense for now, although I agree that the approach could likely be improved in future.

> Please ensure this has some ability to disable at the instance level, so it does not have to show.

@farhaanbukhsh [Your comment above](https://github.com/openedx/platform-roadmap/issues/392#issuecomment-2538791706) suggests that users will be able to disable at an instance level, but I just wanted your confirmation in case the subsequent conversations affected the approach. Is it still the case?

**Comment by Cup0fCoffee** [comment]
> @farhaanbukhsh https://github.com/openedx/platform-roadmap/issues/392#issuecomment-2538791706 suggests that users will be able to disable at an instance level, but I just wanted your confirmation in case the subsequent conversations affected the approach. Is it still the case?

@ali-hugo That was the initial proposal, but the subsequent conversations steered the solution towards adding the option for disabling the "Dates tab" via *Pages and Resources*, which currently only allows changing the settings on per course level.

**Comment by jristau1984** [comment]
> > [@farhaanbukhsh](https://github.com/farhaanbukhsh) [#392 (comment)](https://github.com/openedx/platform-roadmap/issues/392#issuecomment-2538791706) suggests that users will be able to disable at an instance level, but I just wanted your confirmation in case the subsequent conversations affected the approach. Is it still the case?
> 
> [@ali-hugo](https://github.com/ali-hugo) That was the initial proposal, but the subsequent conversations steered the solution towards adding the option for disabling the "Dates tab" via _Pages and Resources_, which currently only allows changing the settings on per course level.

I believe the confirmation is around whether the entire widget wiki be behind a feature flag,  for an instance that does not want this to show on the P&R page at all...

**Comment by marcotuts** [comment]
👍 

@jristau1984  - I'm not sure I agree with you that the Pages & Resources view is on its way to being the Advanced Settings page that seems a long way off. (Though if you have any interest in that topic and can get  👀 on this set of feature ideas proposals we could use feedback - https://openedx.atlassian.net/wiki/spaces/COMM/pages/5236785155/Relocating+Advanced+Settings) 

I don't know how the other tools (Proctoring, Wiki, Calculator) are controlled at the instance level for whether or not they show in courses. Ideally all these tools could be configured at higher levels instead of just the course level. I imagine not all of these tools have the same configurations given when they were added but +1 to adding Dates here regardless.

**Comment by ali-hugo** [comment]
> https://github.com/openedx/platform-roadmap/issues/392#issuecomment-2538791706 suggests that users will be able to disable at an instance level, but I just wanted your confirmation in case the subsequent conversations affected the approach. Is it still the case?
>> @ali-hugo That was the initial proposal, but the subsequent conversations steered the solution towards adding the option for disabling the "Dates tab" via Pages and Resources, which currently only allows changing the settings on per course level.
>>> I believe the confirmation is around whether the entire widget wiki be behind a feature flag, for an instance that does not want this to show on the P&R page at all...


@Cup0fCoffee In addition to the per-course option to disable the **Dates tab** via _Pages and Resources_, would it be possible to disable the **Dates card** at the instance level, so it doesn’t appear on the _Pages and Resources_ page for any course in the instance?

**Comment by ali-hugo** [comment]
@Cup0fCoffee Friendly ping on my comment above. ⏫

**Comment by ali-hugo** [comment]
@Cup0fCoffee Friendly ping on my friendly ping above 😉

**Comment by Cup0fCoffee** [comment]
@ali-hugo Oh, sorry for missing the ping, I was sure I replied to it. Yes, it would be possible to add a flag that would be read by the MFE to toggle the visibility of the **Dates card** on the *Pages and Resources* page.

**Comment by ali-hugo** [comment]
@Cup0fCoffee Great! We can consider this approved then.

**Comment by ali-hugo** [comment]
@Cup0fCoffee One more thing - I'm not sure what the status of this ticket should be e.g. _resourced_, _not resourced_ etc. Could you please apply the correct status? Thanks!

<img width="236" height="249" alt="Image" src="https://github.com/user-attachments/assets/c7aa640d-9370-4f4a-b51b-6a5534f02b66" />

**Comment by Cup0fCoffee** [comment]
@ali-hugo Probably "Not resourced", the particular client that is interested in this only has budget for a specific portion of this feature. However, I know there are people from a mentoring program who are looking to contribute to the platform, so may be someone would be interested in picking this up.

**Comment by ali-hugo** [comment]
@Cup0fCoffee Thanks. I've updated the status.

**Comment by Anas12091101** [comment]
Since no one else has started working on this yet, I’ll proceed with implementing the functionality approved in the discussion above on behalf of the MIT ODL team.

**Comment by ali-hugo** [comment]
@Anas12091101 Thank you to you and the MIT ODL team! I'm happy to hear this feature will be implemented. 

FYI: @Cup0fCoffee

**Comment by Anas12091101** [comment]
Could someone please help review and merge this PR: https://github.com/openedx/openedx-platform/pull/37923

**Comment by sarina** [comment]
https://github.com/openedx/openedx-platform/pull/37923 is merged so this can be closed!
